### DISCUSSION TITLE
AioRTC - Send Image and Receive Image in different applications.

### DISCUSSION THREAD

**Discussion by jamarichal-vis** [aiortc/aiortc#699] [Q&A]
Hello, 
I have seen some examples on the aiortc repo and they have not been able to help me with what I need.

My intention is to make a python application that sends images through a stream. The protocol I want to use is WebRTC, and more specifically the aioRTC implementation. Then, I want a frontend (html + javascript) to receive this streaming and display it on a display. 

My problem is that the examples I have seen in the repository, send the image from the javascript code to the python server, modify the image and send it back. What I want to do is to be able to send the images from python application, not receive them, modify them and send them back. Is this possible through aioRTC?

In conclusion, I want to make an application with aioRTC + opencv, that sends the images I want through the webRTC protocol. Then, I want a client to consume from this application. 

Is it possible to separate the sending of data through aioRTC and the reception of data from a frontend? all the examples I've seen put them together and I don't know if it's possible. 

Do you know if there is any example, that shows in a script the implementation of an aiorct sender and another script that implements the aioRTC receiver?

Regards,

**Comment by osscodee** [comment]
Hi, 

I am a webrtc beginner myself, but from my understanding, in your frontend you add to your peerconnection a transceiver with the direction 'recvonly', otherwise, when you end on the python side the stream, it will fail to create the answer for the offer (when the offer is created by the frontend, e.g. when it is loaded...

 `yourPeerConncetion.addTransceiver('video', { direction: 'recvonly'});`
 
On the python side, before you setRemoteDiscription and createAnswer, you can just add an instance of MediaStreamTrack to your python peer connection. I added the relay, like in the examples, when a stream was "multiplied". 
 
 `backend_python_peerconnection.addTransceiver(relay.subscribe(some_media_stream))`
 
 I think an example implementation of the media stream track is at the bottom of the mediastreams.py located in src/aiortc/mediastreams.py
